nucypher/policy/conditions/lingo.py

```diff
@@ -111,9 +129,25 @@ def __init__(
             "operands": [CONDITION*]
         }
         """
-        self.condition_type = condition_type
+        if condition_type != self.CONDITION_TYPE:
+            raise InvalidCondition(
+                f"{self.__class__.__name__} must be instantiated with the {self.CONDITION_TYPE} type."
+            )
+
         if operator not in self.OPERATORS:
-            raise InvalidLogicalOperator(f"{operator} is not a valid operator")
+            raise InvalidCondition(f"{operator} is not a valid operator")
+
+        if operator == self.NOT_OPERATOR:
+            if len(operands) != 1:
+                raise InvalidCondition(
+                    f"Only 1 operand permitted for '{operator}' condition"
+                )
+        elif len(operands) < 2:
+            raise InvalidCondition(
+                f"Minimum of 2 operand needed for '{operator}' compound condition"
+            )
```

Comment:
This block is similar to `validate_operator_and_operands()` above. Can this be reused?

Reply:
✔️